File: 2780_minimum_index_of_a_valid_split.py
class Solution(object):
    def minimumIndex(self, z):
        from collections import Counter
        n = len(z)
        c = Counter(z)
        for k, v in c.items():
            if v * 2 > n:
                d, t = k, v
                break
        x = 0
        for i in range(n - 1):
            if z[i] == d:
                x += 1
            if x * 2 > i + 1 and (t - x) * 2 > n - i - 1:
                return i
        return -1

File: 2780_minimum_index_of_a_valid_split.cpp
class Solution {
public:
    int minimumIndex(vector<int>& z) {
        int x = 0, y = 0, m = z.size();
        unordered_map<int, int> f;
        for (int v : z) f[v]++;
        for (auto& [k, v] : f) if (v * 2 > m) x = k, y = v;
        int l = 0;
        for (int i = 0; i < m - 1; i++) {
            if ((l += z[i] == x) * 2 > i + 1 && (y - l) * 2 > m - i - 1) return i;
        }
        return -1;
    }
};
